### PR TITLE
Make componentt::{base,pretty}_name comments

### PR DESCRIPTION
**Pull request by tautschnig**
The only semantically relevant name information is the ID_name entry.
Two components should not be considered different when they only differ
in their base_name or pretty_name. Thus, use ID_C_base_name and
ID_C_pretty_name, respectively, to store these.

The goto binary version is incremented as goto binaries compiled before
this patch are incompatible with the changes introduced here.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/goto-programs/write_goto_binary.h [hunk 1]
```diff
@@ -12,7 +12,7 @@ Author: CM Wintersteiger
 #ifndef CPROVER_GOTO_PROGRAMS_WRITE_GOTO_BINARY_H
 #define CPROVER_GOTO_PROGRAMS_WRITE_GOTO_BINARY_H
 
-#define GOTO_BINARY_VERSION 5
+#define GOTO_BINARY_VERSION 6
```
Comment (martin-cs) [inline src/goto-programs/write_goto_binary.h:15]: Does this need a major or minor version number bump?

**Review by martin-cs — APPROVED** [review]
Changes are fine but a change to the format number feels like it is worth a major or minor version number change.

File: src/goto-programs/write_goto_binary.h (hunk 1 above)
Comment (tautschnig) [reply src/goto-programs/write_goto_binary.h:15]: I don't know how many customers have use cases where they build goto binaries, store them, and (re-)analyse them at a later point in time. I know that's something I occasionally see, but I have no idea how common this is. So I'm with you that a version number change is desirable, but I'd lean towards minor-version-bump-is-sufficient. If others agree with this, then we could contemplate merging this PR just before one of the next regular version bumps (e.g., next Thursday).

**Comment by hannes-steffenhagen-diffblue** [comment]
I concur with @martin-cs, changing the goto binary format should be a major version bump as it's a pretty clear example of a breaking change.

While hopefully this would not impact too many people, this gives us an opportunity to tell people "Here, this breaks this particular use case, watch out for that, if you're not doing that you should be fine". 

Of course in principle any change is potentially a breaking change for someone, but my view has been that if we _know_ something is a breaking change it should result in a major version bump.

**Comment by tautschnig** [comment]
As side effect, this PR also sorts out the goto binary version bump that we will want anyhow with version 6 (and the breaking changes that we intend to merge as part of this).

**Comment by kroening** [comment]
> As side effect, this PR also sorts out the goto binary version bump that we will want anyhow with version 6 (and the breaking changes that we intend to merge as part of this).

I haven't seen yet a convincing reason to do this -- I'd be convinced by the changes to the array type, but that needs more work.

**Review by peterschrammel — APPROVED** [review]

**Pull request merged**